src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,282 @@
+        return _next_breaking_change_version()
+
+
+# pylint: disable=too-few-public-methods
+class ExactVersion(TargetVersion):
+    def __init__(self, version):
+        self.version = version
+
+    def __str__(self):
+        return f'in {self.version}'
+
+    def version(self):
+        return self.version()
+
+
+# pylint: disable=too-few-public-methods
+class UnspecificVersion(TargetVersion):
+    def __str__(self):
+        return 'in future'
+
+    def version(self):
+        return None
+
+
+class BreakingChange(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def message(self):
+        pass
+
+    @property
+    @abc.abstractmethod
+    def target_version(self):
+        pass
+
+
+class AzCLIRemoveChange(BreakingChange):
+    """
+    Remove the command groups, commands or arguments in a future release.
+
+    **It is recommended to utilize `deprecate_info` instead of this class to pre-announce Breaking Change of Removal.**
+    :param target: name of the removed command group, command or argument
+    :param target_version: version where the breaking change is expected to happen.
+    :type target_version: TargetVersion
+    :param redirect: alternative way to replace the old behavior
+    :param doc_link: link of the related document
+    """
+    def __init__(self, target, target_version=NextBreakingChangeWindow(), redirect=None, doc_link=None):
+        self.target = target
+        self._target_version = target_version
+        self.alter = redirect
+        self.doc_link = doc_link
+
+    @property
+    def message(self):
+        alter = f' Please use {self.alter} instead.' if self.alter else ''
+        doc = f' To know more about the Breaking Change, please visit {self.doc_link}.' if self.doc_link else ''
+        return f'`{self.target}` will be removed {str(self._target_version)}.{alter}{doc}'
+
+    @property
+    def target_version(self):
+        return self._target_version
+
+
+class AzCLIRenameChange(BreakingChange):
+    """
+    Rename the command groups, commands or arguments to a new name in a future release.
+
+    **It is recommended to utilize `deprecate_info` instead of this class to pre-announce Breaking Change of Renaming.**
+    It is recommended that the old name and the new name should be reserved in few releases.
+    :param target: name of the renamed command group, command or argument
+    :param new_name: new name
+    :param target_version: version where the breaking change is expected to happen.
+    :type target_version: TargetVersion
+    :param doc_link: link of the related document
+    """
+    def __init__(self, target, new_name, target_version=NextBreakingChangeWindow(), doc_link=None):
+        self.target = target
+        self.new_name = new_name
+        self._target_version = target_version
+        self.doc_link = doc_link
+
+    @property
+    def message(self):
+        doc = f' To know more about the Breaking Change, please visit {self.doc_link}.' if self.doc_link else ''
```

Comment:
This sentence seems to have appeared many times. Could we abstract it into a reusable constant for future maintenance and updates

Reply:
Done